### DISCUSSION TITLE
Command-line interface

### DISCUSSION THREAD

**Discussion by nside** [protomaps/protomaps.js#101] [General]
Hi,
I'd like to hear your opinion about packaging this as a command-line tool that can render map PNGs. Either by tile coordinates or arbitrary bounding boxes. How could it be done with minimal effort?

**Comment by bdon** [comment]
I don't think it's a good fit for that use case. Doing so would require running JS and a Canvas rendering context on the server, which would require an entire headless web browser. The focus of the protomaps.js renderer is lightweight rendering.

The two other solutions for which infrastructure already exists are: 
1) run [Mapnik](https://github.com/mapnik) on the server
2) run [MapLibre GL native](https://github.com/maplibre/maplibre-gl-native) headlessly on the server
